Zend/zend_compile.h

```diff
@@ -517,6 +518,7 @@ typedef struct _zend_internal_function {
 
 	zif_handler handler;
 	struct _zend_module_entry *module;
+	const zend_frameless_function_info *frameless_function_infos;
```

Comment:
Instead of having few different handlers for different aritirs, we may have only a single handler for the biggest possible arity. Then during compilation we should add the necessary default constant arguments.

This is just an idea. I'm not sure if it may simplify things or would make them even more complex.

Reply:
Arity-specialization might lead to slightly more efficient handlers (i.e. dropped null/false checks). Your suggestion seems useful for functions that don't need a high degree of specialization or that have a lot of default parameters.

Reply:
Implemented, thanks for the nice suggestion!